### DISCUSSION TITLE
RLS Authorized user view/update own records

### DISCUSSION THREAD

**Discussion by jwright04** [supabase/supabase#7547] [Questions] [answered]
I seem to be having an issue understanding RLS on Supabase.  I have an app that allows a single user to perform CRUD operations on the data that they post.  I've enabled RLS which is now causing errors (which I understand why, just not sure how to fix).  After reading through the documentation
https://supabase.com/docs/guides/auth/row-level-security

I've created a policy from a template "Enable insert for authenticated users only: true" 

My policy name is
"Enable insert for authenticated users only"

Target roles
"Authenticated"

WITH CHECK expression is
"auth.uid() = id" (also tried with (uid() = id))

Returns with a toast error :
"Error updating policy: failed to update pg.policies with the given ID: operator does not exist: uuid = bigint"

I feel like I'm missing something really simply but not sure.  Can someone lend a hand?

**Comment by burggraf** [comment, marked as answer]
It sounds like the `id` field you are referencing here is not of type `uuid`.  Did you make it a `bigserial` or `bigint` instead?

You can't compare a `uuid` i.e. `auth.uid()` and a `bigint`.

**Reply by jwright04** [reply]
Thanks for the tip, I ended up using `auth.uid() = user_id` and it worked for me.
